pkgs/development/tools/rust/rust-analyzer/generic.nix

```diff
@@ -17,6 +17,24 @@ rustPlatform.buildRustPackage {
     inherit rev sha256;
   };
 
+  patchFlags = [ "-Rp1" ];
+  depsExtraArgs.patchFlags = [ "-Rp1" ];
+  cargoPatches = [
+    # Patches required before rust 1.51.0 to revert commits updating arrayvec.
+    (fetchpatch { # cargo update
+      url = "https://github.com/rust-analyzer/rust-analyzer/commit/c7a664c2ea80cb8d7fa94faf7468210b1f109b60.patch";
+      sha256 = "sha256-ZQdIDvvdBSanAPPVzd7+UJgJ4j6Mp6KFhyALCoD6n2A=";
+    })
+    (fetchpatch { # Use more std::array::IntoIter
+      url = "https://github.com/rust-analyzer/rust-analyzer/commit/bc5c86543b1bf384555471415dee770888a1781d.patch";
+      sha256 = "sha256-IfPe8dx2UQO+zMZ8u3qzVqDeyfcMhq46ZKEMQvDCzXA=";
+    })
+    (fetchpatch { # Use arrayvec 0.6
+      url = "https://github.com/rust-analyzer/rust-analyzer/commit/9787bddac577a6aa24388fb91286474a7a8cf0bc.patch";
+      sha256 = "sha256-pypbzMNJ8ywUp4pA+UQ3dsqrg2tVf0bNt4CE22w7aVk=";
+    })
+  ];
```

Comment:
Those patches are already included in 2021-04-05. Do I miss something?

Reply:
> Patches required before rust 1.51.0 to revert commits updating arrayvec.

Currently we have 1.50.0 in nixpkgs master, but these changes made it requires 1.51.0 to compile (1.51.0 is still in staging). This blocks rust-analyzer to some weeks. Here we **revert** some commits to made it compile on 1.50.0.

Reply:
They are now removed for rustc 1.51.0 in master.